src/agent.rs

```diff
@@ -345,21 +538,53 @@ impl Agent for CodexAgent {
             .map(|m| m.current_mode_id.clone())
             .unwrap_or(acp::SessionModeId("auto".into()));
 
-        let session_id = conversation_id.to_string();
-        // Populate a placeholder session entry prior to spawning Codex so that
-        // pipelined requests (like an immediate `/status`) see the session.
         self.sessions.borrow_mut().insert(
             session_id.clone(),
             SessionState {
+                conversation_id: String::new(),
+                conversation: None,
                 current_approval: self.config.approval_policy,
                 current_sandbox: self.config.sandbox_policy.clone(),
-                current_mode,
+                current_mode: current_mode.clone(),
                 token_usage: None,
                 reasoning_sections: Vec::new(),
                 current_reasoning_chunk: String::new(),
             },
         );
 
+        let session_config = match self.build_session_config(&session_id) {
+            Ok(cfg) => cfg,
+            Err(err) => {
+                self.sessions.borrow_mut().remove(&session_id);
+                return Err(err);
+            }
+        };
+
+        let new_conv = self
+            .conversation_manager
+            .new_conversation(session_config)
+            .await;
+
+        let (conversation, session_configured) = match new_conv {
+            Ok(NewConversation {
+                conversation,
+                session_configured,
+                ..
+            }) => (conversation, session_configured),
+            Err(e) => {
+                warn!(error = %e, "Failed to create Codex conversation");
+                self.sessions.borrow_mut().remove(&session_id);
+                return Err(Error::into_internal_error(e));
+            }
+        };
+
+        if let Ok(mut sessions) = self.sessions.try_borrow_mut()
+            && let Some(state) = sessions.get_mut(&session_id)
+        {
+            state.conversation_id = session_configured.session_id.to_string();
```

Comment:
1. why use `session_configured.session_id` as `conversation_id` rather than `NewConversation.conversation_id`?
2. can you move conversation initialize to L540?